test/unit/gateways/trans_first_transaction_express_test.rb

```diff
@@ -257,6 +257,15 @@ def test_empty_response_fails
     assert_equal nil, response.message
   end
 
+  def test_transaction_code_xml_tag_added_with_correct_prefix
+    stub_comms do
+      @gateway.purchase(@amount, @credit_card)
+    end.check_request do |_endpoint, data, _headers|
+      doc = Nokogiri::XML(data)
+      assert_not_empty doc.xpath('//v1:tranCode', 'v1' => 'http://postilion/realtime/merchantframework/xsd/v1/')
```

Comment:
Does this create a sample purchase and check the formatted post request for the xpath?

Reply:
Yes.

In the `check_request` block the `data` variable holds the request we send to the gateway. So this test is just showing that there is a non-empty `tranCode` field with a `v1` prefix in the request.

If you run this test with Nokogiri 1.13.4 and my changes in the gateway adapter commented-out it will fail.